scot/csp.py

```diff
@@ -59,7 +59,7 @@ def csp(x, cl, numcomp=None):
         sigma2 += np.cov(x2[t, :, :]) / x2.shape[0]
     sigma2 /= sigma2.trace()
 
-    e, w = eig(sigma1, sigma1 + sigma2, overwrite_a=True, overwrite_b=True, check_finite=False)
+    e, w = eigh(sigma1, sigma1 + sigma2, overwrite_a=True, overwrite_b=True, check_finite=False)
```

Comment:
Because we're dealing with Hermitian matrices. If our matrices are not Hermitian, this indicates a problem with our covariance matrices (e.g. due to a lack of data and/or no regularization). In such cases, `eigh` fails and tells us that something is wrong. `eig` on the other hand silently produces complex eigenvalues, which we usually ignored. For example, there was a problem with one test case, which generated a sine as input signal.